src/subapps/projects/views/ActivityView.tsx

```diff
@@ -167,6 +171,41 @@ const ActivityView: React.FC = () => {
     setRefreshActivities(!refreshActivities);
   };
 
+  const linkCodeToActivity = (codeResourceId: string) => {
+    nexus.Resource.getSource(
+      orgLabel,
+      projectLabel,
+      encodeURIComponent(activityId)
+    )
+      .then(response => {
+        const payload = response as ActivityResource;
+
+        if (payload.wasAssociatedWith) {
+          payload.wasAssociatedWith = Array.isArray(payload.wasAssociatedWith)
+            ? [...payload.wasAssociatedWith, { '@id': codeResourceId }]
+            : [payload.wasAssociatedWith, { '@id': codeResourceId }];
+        } else {
+          payload.wasAssociatedWith = { '@id': codeResourceId };
+        }
+
+        activity &&
+          nexus.Resource.update(
```

Comment:
Don't nest your .then's and .catch's, but instead return the promise, and add another then() statement on the end of the main one.